core/src/test/java/org/apache/druid/timeline/partition/DimensionRangeShardSpecTest.java

```diff
@@ -297,6 +301,98 @@ public void testIsInChunk_withMultiValues()
     ));
   }
 
+  @Test
+  public void testPossibleInDomain()
```

Comment:
If this test method seems to be getting too big, you could split it up into two, one each for positive cases and negative cases.

Reply:
I second this. Currently `assertFalse` and `assertTrue`'s are mixed up. One does have to look closely to get the test case. Can it also help us in drawing parallels among the testcases? (for many True case there should be a counter False case as well)